oauth-server-plain/src/main/java/io/strimzi/kafka/oauth/server/plain/JaasServerOauthOverPlainValidatorCallbackHandler.java

```diff
@@ -0,0 +1,124 @@
+
+        if (!"PLAIN".equals(saslMechanism))    {
+            throw new IllegalArgumentException(String.format("Unexpected SASL mechanism: %s", saslMechanism));
+        }
+
+        ServerConfig config = parseJaasConfig(jaasConfigEntries);
+
+        String tokenEndpoint = config.getValue(ServerPlainConfig.OAUTH_TOKEN_ENDPOINT_URI);
+        if (tokenEndpoint == null) {
+            throw new IllegalArgumentException("tokenEndpointUri == null");
+        }
+        try {
+            this.tokenEndpointUri = new URI(tokenEndpoint);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("Invalid tokenEndpointUri: " + tokenEndpoint, e);
+        }
+
+        super.configure(configs, "OAUTHBEARER", jaasConfigEntries);
+
+        log.debug("Configured OAuth over PLAIN:\n    tokenEndpointUri: " + tokenEndpointUri);
+    }
+
+    @Override
+    public void close() {
+        super.close();
+    }
+
+    @Override
+    public void handle(Callback[] callbacks) throws UnsupportedCallbackException {
+        String username = null;
+        String password = null;
+        org.apache.kafka.common.security.plain.PlainAuthenticateCallback cb = null;
+
+        for (Callback callback : callbacks) {
+            if (callback instanceof javax.security.auth.callback.NameCallback) {
+                username = ((javax.security.auth.callback.NameCallback) callback).getDefaultName();
+            } else if (callback instanceof org.apache.kafka.common.security.plain.PlainAuthenticateCallback) {
+                password = String.valueOf(((org.apache.kafka.common.security.plain.PlainAuthenticateCallback) callback).password());
+                cb = (org.apache.kafka.common.security.plain.PlainAuthenticateCallback) callback;
+            } else {
+                throw new UnsupportedCallbackException(callback);
+            }
+        }
+
+        handleCallback(cb, username, password);
+    }
+
+    private void handleCallback(PlainAuthenticateCallback callback, String username, String password) {
+        if (callback == null) throw new IllegalArgumentException("callback == null");
+        if (username == null) throw new IllegalArgumentException("username == null");
+
+        try {
+            authenticate(username, password);
+            callback.authenticated(true);
+
+        } catch (SaslAuthenticationException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new SaslAuthenticationException("Authentication failed: " + getCauseMessage(e), e);
+        }
+    }
+
+    private void authenticate(String username, String password) throws UnsupportedCallbackException, IOException {
+        // if username equals 'access-token' we treat the password as an access token
+        // otherwise, we treat username as clientId, password as client secret and perform client credential auth
+        // to get the access token
+        final String accessToken;
+        if ("access-token".equals(username)) {
```

Comment:
I wonder if we should give this name a bit more special feeling. for example `$accessToken` This fits with what for example Microsoft does on EventHubs. I think it makes it a bit more clear that it is some special value with special meaning when reading the different configuration files using it.

Reply:
Makes sense, but for this specific suggestion of `$accessToken` my concern is that it would cause problems in the CLI use-cases. E.g.:

    kafkacat -b my-cluster-kafka-bootstrap:9092 -X security.protocol=SASL_PLAINTEXT -X sasl.mechanism=PLAIN -X sasl.username=$accessToken -X sasl.password="$ACCESS_TOKEN" -t a_topic -P

This will result in empty username, and authentication will fail. For that reason I would not use $ in the reserved word.
But maybe we could use @ as in `@accessToken`.

Reply:
MS is using `$` ... that was why I suggested it. But you have a good point, although I'm not sure how often will people really use it from the command line. 

@strimzi/maintainers WDYT?.

Reply:
I don't have a strong opinion. If it's open to interpolation by the shell it can be enclosed in `'`.

Reply:
Ok, so I guess '$accessToken' seems more natural to most people, and they would understand this as setting a kind of server side variable rather than refer to the client side environment variable. And escaping it in the shell is simple.